driver/pace/driver/state.py

```diff
@@ -175,12 +179,116 @@ def _overwrite_state_from_restart(
+):
+    """
+    Takes a Pace driver state
+    and returns a dict of state quantities with their Fortran names
+    """
+
+    dict_state = {}
+
+    for _field in fields(type(driver_state)):
+        if "units" in _field.metadata.keys():
+            if is_gpu_backend:
+                if "phy" in restart_file_prefix:
+                    dict_state[_field.name] = driver_state.__dict__[_field.name][:]
+                else:
+                    dict_state[_field.name] = driver_state.__dict__[_field.name].data[:]
+            else:
+                dict_state[_field.name] = driver_state.__dict__[_field.name].data[:]
+
+    return dict_state
+
+
+def _dict_state_to_driver_state(
+    fortran_state: dict,
+    driver_state: Union[fv3core.DycoreState, pace.physics.PhysicsState, TendencyState],
+    is_gpu_backend: bool,
+):
+    """
+    Takes a dict of state quantities with their Fortran names and a driver state
+    and populates the driver state with quantities from the dict.
+    """
+
+    for _field in fields(type(driver_state)):
+        if "units" in _field.metadata.keys():
+            if is_gpu_backend:
+                if (
+                    _field.name in fortran_state
+                    and _field.name in driver_state.__dict__.keys()
+                ):
+                    if isinstance(
+                        driver_state, pace.physics.physics_state.PhysicsState
+                    ):
+                        driver_state.__dict__[_field.name][
+                            halo : -halo - 1, halo : -halo - 1, :-1
+                        ] = gt_utils.asarray(
+                            np.transpose(fortran_state[_field.name].data),
+                            to_type=cp.ndarray,
+                        )
+                    else:
+                        driver_state.__dict__[_field.name].view[:] = gt_utils.asarray(
+                            np.transpose(fortran_state[_field.name].data),
+                            to_type=cp.ndarray,
+                        )
+            else:
+                if (
+                    _field.name in fortran_state
+                    and _field.name in driver_state.__dict__.keys()
+                ):
+                    if isinstance(
+                        driver_state, pace.physics.physics_state.PhysicsState
+                    ):
+                        driver_state.__dict__[_field.name][
+                            halo : -halo - 1, halo : -halo - 1, :-1
+                        ] = np.transpose(fortran_state[_field.name].data)
+                    else:
+                        driver_state.__dict__[_field.name].view[:] = np.transpose(
+                            fortran_state[_field.name].data
+                        )
+
+    return driver_state
+
+
 def _restart_driver_state(
     path: str,
     rank: int,
     quantity_factory: pace.util.QuantityFactory,
     communicator: pace.util.CubedSphereCommunicator,
+    fortran_data: bool = False,
```

Comment:
I think you can delete this argument by making `_overwrite_state_from_restart` dynamically detect whether the path is a fortran or pace restart folder.